test/integration/query-interface/changeColumn.test.js

```diff
@@ -361,6 +362,131 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         expect(refs[0].referencedTableName).to.equal('Users');
         expect(refs[0].referencedColumnName).to.equal('id');
       });
+
+      it('should retain ON UPDATE and ON DELETE constraints after a column is changed', async function () {
+        await this.queryInterface.createTable('users', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          name: {
+            type: DataTypes.STRING,
+            allowNull: true,
+          },
+          level_id: {
+            type: DataTypes.INTEGER,
+            allowNull: false,
+            references: {
+              key: 'id',
+              model: 'level',
+            },
+            onDelete: 'CASCADE',
+            onUpdate: 'CASCADE',
+          },
+        });
+
+        await this.queryInterface.createTable('level', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          name: {
+            type: DataTypes.CHAR,
+            allowNull: false,
+          },
+        });
+
+        await this.queryInterface.changeColumn('users', 'name', {
+          type: DataTypes.CHAR,
+          allowNull: false,
+        });
+
+        const constraintsQuery = this.queryInterface.queryGenerator.showConstraintsQuery('users');
+        const [{ sql: usersSql }] = await this.queryInterface.sequelize.query(constraintsQuery, {
+          type: 'SELECT',
+        });
+
+        expect(usersSql).to.include('ON DELETE CASCADE', 'should include ON DELETE constraint');
+        expect(usersSql).to.include('ON UPDATE CASCADE', 'should include ON UPDATE constraint');
+      });
+
+      it('should change columns with foreign key constraints without data loss', async function () {
+        await this.queryInterface.createTable('users', {
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          name: {
+            type: DataTypes.STRING,
+            allowNull: true,
+          },
+          level_id: {
+            type: DataTypes.INTEGER,
+            allowNull: false,
+            references: {
+              key: 'id',
+              model: 'level',
+            },
+            onDelete: 'CASCADE',
+            onUpdate: 'CASCADE',
+          },
+        });
+
+        await this.queryInterface.createTable('level', {
```

Comment:
`level` should be created first, it is needed by `users`

Reply:
Done!